packages/flutter_tools/lib/src/commands/apk.dart

```diff
@@ -151,6 +179,33 @@ class ApkCommand extends FlutterCommand {
         help: 'Password for the entry within the keystore.');
   }
 
+  Future _findServices(_ApkComponents components) async {
+    if (!ArtifactStore.isPackageRootValid)
+      return;
+
+    dynamic manifest = _loadYamlFile(_kFlutterManifestPath);
+    if (manifest['services'] == null)
+      return;
+
+    for (String service in manifest['services']) {
+      String serviceRoot = '${ArtifactStore.packageRoot}/$service/apk';
+      dynamic serviceConfig = _loadYamlFile('$serviceRoot/config.yaml');
+      if (serviceConfig == null || serviceConfig['jars'] == null)
+        continue;
+      components.services.addAll(serviceConfig['services']);
+      for (String jar in serviceConfig['jars']) {
+        // Jar might refer to an android SDK jar, or URL to download.
+        jar = jar.replaceAll('\$androidSdk', '${components.androidSdk.path}');
+        if (jar.startsWith("http")) {
```

Comment:
I wonder if we want to change `$androidSdk` into something like `android-sdk:` so that all the entries in this list are URLs of some form.